src/SFML/Window/DRM/DRMContext.cpp

```diff
@@ -130,6 +128,306 @@ namespace
+        {
+            // Possible_crtcs is a bitmask as described here:
+            // https://dvdhrm.wordpress.com/2012/09/13/linux-drm-mode-setting-api
+            const sf::Uint32 crtcMask = 1U << i;
+            const sf::Uint32 crtcId = resources.crtcs[i];
+            if (encoder.possible_crtcs & crtcMask)
+            {
+                return crtcId;
+            }
+        }
+
+        // No match found
+        return 0;
+    }
+
+    sf::Uint32 findCrtcForConnector(const sf::priv::Drm& drm, const drmModeRes& resources, const drmModeConnector& connector)
+    {
+        for (int i = 0; i < connector.count_encoders; ++i)
+        {
+            const sf::Uint32 encoderId = connector.encoders[i];
+            const drmModeEncoderPtr encoder = drmModeGetEncoder(drm.fd, encoderId);
+
+            if (encoder)
+            {
+                const sf::Uint32 crtcId = findCrtcForEncoder(resources, *encoder);
+
+                drmModeFreeEncoder(encoder);
+                if (crtcId != 0)
+                {
+                    return crtcId;
+                }
+            }
+        }
+
+        // No match found
+        return 0;
+    }
+
+    int getResources(int fd, drmModeResPtr& resources)
+    {
+        resources = drmModeGetResources(fd);
+        if (resources == NULL)
+            return -1;
+        return 0;
+    }
+
+    int hasMonitorConnected(int fd, drmModeRes& resources)
+    {
+        drmModeConnectorPtr connector;
+        for (int i = 0; i < resources.count_connectors; ++i)
+        {
+            connector = drmModeGetConnector(fd, resources.connectors[i]);
+            if (connector->connection == DRM_MODE_CONNECTED)
+            {
+                // There is a monitor connected
+                drmModeFreeConnector(connector);
+                connector = NULL;
+                return 1;
+            }
+            drmModeFreeConnector(connector);
+            connector = NULL;
+        }
+        return 0;
+    }
+
+    int findDrmDevice(drmModeResPtr& resources)
+    {
+        static const int maxDrmDevices = 64;
+
+        drmDevicePtr devices[maxDrmDevices] = { NULL };
+
+        const int numDevices = drmGetDevices2(0, devices, maxDrmDevices);
+        if (numDevices < 0)
+        {
+            sf::err() << "drmGetDevices2 failed: " << std::strerror(-numDevices) << std::endl;
+            return -1;
+        }
+
+        int fd = -1;
+        for (int i = 0; i < numDevices; ++i)
+        {
+            drmDevicePtr device = devices[i];
+            int ret;
+
+            if (!(device->available_nodes & (1 << DRM_NODE_PRIMARY)))
+                continue;
+            // OK, it's a primary device. If we can get the drmModeResources, it means it's also a KMS-capable device.
+            fd = open(device->nodes[DRM_NODE_PRIMARY], O_RDWR);
+            if (fd < 0)
+                continue;
+            ret = getResources(fd, resources);
+            if(std::getenv("SFML_DRM_DEBUG"))
```

Comment:
I'm not a fan of using an environment variable to dynamic enable or disable debug logs. I want to change this to instead use the `SFML_DEBUG` preprocessor symbol. That way we stop doing so many environment variable lookups.